packages/docs/celo-codebase/protocol/optics.md

```diff
@@ -0,0 +1,180 @@
+
+## Optical Channels for Cross-Chain Communication
+
+Optics sends messages from one chain to another in the form of raw bytes. A cross-chain application that wishes to use Optics will need to define the rules for sending and receiving messages for its use case.
+
+Each cross-chain application must implement its own messaging protocol. By convention, we call the contracts that implement this protocol the application's Router contracts. Their function is broadly similar to routers in local networks. They ensure that incoming and outgoing messages are in the protocol-defined format, and facilitate handling and dispatch.
+
+__These Router contracts must:__
+
+- maintain a permissioned set of the contract(s) on remote chains from which it will accept messages via Optics — this could be a single owner of the application on one chain; it could be a registry of other applications implementing the same rules on various chains
+- encode messages in a standardized format, so they can be decoded by the Router contract on the destination chain
+- handle messages from remote Router contracts
+- dispatch messages to remote Router contracts
+
+By implementing these pieces of functionality within a Router contract and deploying it across multiple chains, we create a working cross-chain application using a common language and set of rules. Applications of this kind may use Optics as the cross-chain courier for sending and receiving messages to each other.
+
+## Benefits and Trade-offs of the Optics Architecture
+
+__Benefits__ | __Challenges__
+-- | --
+Broadcast channel | 1-of-N fraud-proof trust model
+Cheap | Liveness failure if updater goes down
+One-solution fits all | Updater could voluntarily forfeit bond to try to pass forged messages
+Simple |  
+
+The benefit of Optics is that the broadcast channel allows for a single-producer, multi-consumer model. This ensures that 1 accumulator can communicate with any number of receiving chains. It’s also much cheaper than other options, allowing updates and proofs to cost <100k gas and be checked by only 1 signature. With Optics, one-solution fits all meaning that constraints on receiving chains are minimal (1 hash function + 1 signature check). There is no implementation or security difference between Proof of Stake and Proof of Work chains.  There is also many fewer LoC than a Relay, much lower design maintenance overhead, and much less expertise required to maintain and operate.
+
+We’ve been careful to address all concerns with the Optics system and have designed solutions that allow for optimal speed, cost, and security of the network. For example, we rely on fraud publication rather than fraud proofs to improve the speed and cost of sending messages. In this security model, any potential fraud is disincentivized and costly, and all participants will always learn of any potential fraud with plenty of time to mitigate harm.
+
+
```

Comment:
Are there links to the code implementation? If not, we should add them here

Reply:
Not yet. We plan to iterate to include examples in the near future.